ext/mbstring/mbstring_singlebyte.h

```diff
@@ -0,0 +1,44 @@
+/*
```

Comment:
Why is this not located inside libmbfl? It also looks like this is not in the windows build system, thus the AppVeyor build failure.

Reply:
Don't know how I missed the Windows build file. Thanks.

I feel the separation between `mbstring` and `libmbfl` is not helping anything, and it would be simpler to move everything into `mbstring`.

If you have a different view, please mention.

Reply:
I do think we should keep all the filter implementations in one place, which is currently `ext/mbstring/libmbfl/filters`. It's already confusing enough that we have *some* filters located in ext/mbstring/libmbfl/mbfl instead (e.g. the 8bit filter -- why?)

Reply:
Sure, good point.

Would it change your mind if I told you that I am hoping to do something similar with the other filters? Right now, there is a very great deal of duplication in all the legacy CJK encoding code, and I expect combining them will lead to a very nice reduction in LOC.

I had considered whether we should combine all the SJIS variants together, the EUC variants together, and so on... but the thing is: SJIS, EUC-JP, and ISO-2022-JP all encode characters from the JIS X 0208 charset, so I expect there will be a non-trivial amount of repeated code which can be squeezed out of those. Again, SJIS-2004, EUC-JP-2004, and ISO-2022-JP-2004 all encode characters from the same charset, so again I expect there will be significant duplication.

Again, ISO-2022-KR uses a different charset from ISO-2022-JP, but the code for handling escapes should be very similar.

After that, hardly anything will be left. I might combine HTML entity encoding/uuencoding/qprint encoding/"8 bit" encoding, etc. In other words, *byte* encodings as opposed to *character* encodings.

I expect that those will squeeze down to almost nothing by reusing functions in the standard library rather than re-implementing them.

Anyways, please just tell me which way you want to go.

Reply:
Even if these are further combined, I still think that the basic separation in the current structure makes sense: There's the generic library portion in libmbfl, and there's the PHP integration in mbstring. As the encoding filters are part of the generic library, I think they should stay in the libmbfl tree.

Reply:
Fair enough.

Reply:
Done.